sympy/solvers/ode/single.py

```diff
@@ -492,59 +499,32 @@ class AlmostLinear(SinglePatternODESolver):
     has_integral = True
 
     def _wilds(self, f, x, order):
-        P = Wild('P', exclude=[f(x)])
+        P = Wild('P', exclude=[f(x).diff(x)])
         Q = Wild('Q', exclude=[f(x).diff(x)])
         return P, Q
 
     def _equation(self, fx, x, order):
-        P,Q = self.wilds()
-        df = fx.diff(x)
-        eq = self.ode_problem.eq_expanded
-        c = Wild('c', exclude=[fx])
-        d = Wild('d', exclude=[df, fx.diff(x, 2)])
-        e = Wild('e', exclude=[df])
-        eq = expand(eq)
-        eq = eq.collect(fx)
-        r = eq.match(e*df+d)
-        self.fxx = None
-        self.gx = None
-        self.kx = None
-        if r:
-            r2 = r.copy()
-            r2[c] = S.Zero
-            if r2[d].is_Add:
-                # Separate the terms having f(x) to r[d] and
-                # remaining to r[c]
-                no_f, r2[d] = r2[d].as_independent(fx)
-                r2[c] += no_f
-            factor = simplify(r2[d].diff(fx)/r[e])
-            if factor and not factor.has(fx):
-                r2[d] = factor_terms(r2[d])
-                u = r2[d].as_independent(fx, as_Add=False)[1]
-                r2.update({'a': e, 'b': d, 'c': c, 'u': u})
-                r2[d] /= u
-                r2[e] /= u.diff(fx)
-                self.coeffs = r2
-                du = u.diff(x)
-                temp = du + (r2[r2['b']]/r2[r2['a']])*u - r2[r2['c']]/r2[r2['a']]
-                temp = expand(temp/ temp.coeff(fx.diff(x)))
-                self.fxx = r2[e]
-                self.kx = r2[d]
-                self.gx = -r2[c]
-                self.substituting = u
-                return fx.diff(x) + P*fx - Q
-        return S.Zero
+        P, Q = self.wilds()
+        self.simplify = False
+        return P*fx.diff(x) + Q
 
-    def _get_general_solution(self, *, simplify: bool = True):
+    def _verify(self):
         P, Q = self.wilds_match()
         fx = self.ode_problem.func
+        R, Q = Q.as_independent(fx) if Q.is_Add else (S.Zero, Q)
+        if Q.diff(fx) != 0 and not simplify(Q.diff(fx)/P).has(fx):
+            self.ly = factor_terms(Q).as_independent(fx, as_Add=False)[1]
+            self.fxx = P / self.ly.diff(fx)
```

Comment:
It's not easy to follow what all of this is doing. It would be better if there were some comments or a docstring and if the symbols had names that were relatable to the forms given in the main docstring.